exporter/loadbalancingexporter/resolver_aws_cloudmap.go

```diff
@@ -29,10 +31,10 @@ var (
 	errNoNamespace   = errors.New("no Cloud Map namespace specified to resolve the backends")
 	errNoServiceName = errors.New("no Cloud Map service_name specified to resolve the backends")
 
-	awsResolverMutator = tag.Upsert(tag.MustNewKey("resolver"), "aws")
-
-	awsResolverSuccessTrueMutators  = []tag.Mutator{awsResolverMutator, successTrueMutator}
-	awsResolverSuccessFalseMutators = []tag.Mutator{awsResolverMutator, successFalseMutator}
+	awsResolverAttr           = attribute.String("resolver", "aws")
+	awsResolverAttrSet        = attribute.NewSet(awsResolverAttr)
+	awsResolverSuccessAttrSet = attribute.NewSet(awsResolverAttr, attribute.Bool("success", true))
+	awsResolverFailureAttrSet = attribute.NewSet(awsResolverAttr, attribute.Bool("success", false))
```

Comment:
To make sure I'm following, each resolver needs its own definition of success and failure as the `resolver` attribute is also included? Would there be any way to move the success attribute to some central definition and then each resolver references it?

Reply:
It's possible to have them centrally defined, but when it's time to consume, we'll end up with new allocations on the hot path. Given that they are static, we can pre-create them here and use them later, without new allocations on the hot path. 

The allocations I have in mind are: 

- new Key, Value and KeyValue for the attributes themselves (can be avoided with the central definition)
- new KeyValue slice for the function call to WithAttributes as part of Record
- new slice for the underlying AttributeSet created when using the "WithAttributes" metric option.

Using WithAttributeSet doesn't make allocations (as far as I could see) and having the AttributeSet created only once per resolver seems the best of the worlds for me.

Reply:
That makes sense, thanks for explaining!